packages/jest-circus/src/run.ts

```diff
@@ -166,8 +166,8 @@ const _callCircusTest = async (
   test: Circus.TestEntry,
   testContext: Circus.TestContext,
 ): Promise<void> => {
-  await dispatch({name: 'test_fn_start', test});
   const timeout = test.timeout || getState().testTimeout;
+  await dispatch({name: 'test_fn_start', test, timeout});
```

Comment:
instead of dispatching with the timeout, we could just mutate `test.timeout`. Not sure if people want to be able to differentiate between "global" timeout and test specific one - I'm guessing "no"